app/src/main/java/com/example/sharingang/database/store/UserStore.kt

```diff
@@ -58,4 +59,57 @@ interface UserStore {
      * @return true if yes, false if no
      */
     suspend fun hasBeenReported(reporterId: String, reportedId: String): Boolean
+
+    /**
+     * Retrieves all the chat partners of a particular user
+     *
+     * @param userId the id of the user whose partners to retrieve
+     * @return the list of all ids of the partners of the user
+     */
+    suspend fun getChatPartners(userId: String): MutableList<String>
+
+    /**
+     * Retrieves all messages of a user with a particular user
+     *
+     * @param userId the user id of the user whose messages we want to retrieve
+     * @param with the target user
+     * @return the resulting list of messages / Chats
+     */
+    suspend fun getMessages(userId: String, with: String): MutableList<Chat>
+
+    /**
+     * Puts a message into database
+     *
+     * @param from the sender
+     * @param to the receiver
+     * @param message the message
+     * @return the new list of messages between the two users
+     */
+    suspend fun putMessage(from: String, to: String, message: String): MutableList<Chat>
+
+    /**
+     * Sets up the listener on messages for a particular user with another user
+     *
+     * @param userId the current user id
+     * @param with the target user id
+     * @param action what to do upon getting notified
+     */
+    suspend fun setupRefresh(userId: String, with: String, action: () -> Unit)
```

Comment:
Would be nice to reference the chat in the name of the method.
Maybe `setupChatRefresh` or `setupConversationRefresh`